File: src/main/java/com/wowo/wowo/domain/user/valueobjects/UserId.java
package com.wowo.wowo.domain.user.valueobjects;

/**
 * User ID value object
 */
public record UserId(String value) {
    
    public UserId {
        if (value == null || value.trim().isEmpty()) {
            throw new IllegalArgumentException("User ID cannot be null or empty");
        }
    }
    
    public static UserId of(String id) {
        return new UserId(id.trim());
    }
    
    @Override
    public String toString() {
        return value;
    }
}

File: src/main/kotlin/com/wowo/wowo/contexts/user/domain/valueobject/UserId.kt
package com.wowo.wowo.contexts.user.domain.valueobject

import com.wowo.wowo.shared.domain.ValueObject
import java.util.*

/**
 * Value Object representing User ID
 */
data class UserId(val value: UUID = UUID.randomUUID()) : ValueObject {
    override fun toString(): String = value.toString()

    companion object {
        fun fromString(id: String): UserId = UserId(UUID.fromString(id))
    }
}


